apps/crn-frontend/src/network/__tests__/EventsEmbed.test.tsx

```diff
@@ -0,0 +1,131 @@
+import { createListEventResponse } from '@asap-hub/fixtures';
+import { Constraint } from '@asap-hub/model';
+import { network } from '@asap-hub/routing';
+import {
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Suspense } from 'react';
+import { MemoryRouter, Route } from 'react-router-dom';
+import { RecoilRoot, RecoilState } from 'recoil';
+import { Auth0Provider, WhenReady } from '../../auth/test-utils';
+import { getEventsFromAlgolia } from '../../events/api';
+import { eventsState } from '../../events/state';
+import { CARD_VIEW_PAGE_SIZE } from '../../hooks';
+import Events from '../EventsEmbed';
+import { refreshTeamState } from '../teams/state';
+import { refreshUserState } from '../users/state';
+
+jest.mock('../../events/api');
+
+afterEach(() => {
+  jest.resetAllMocks();
+});
+
+const mockGetEvents = getEventsFromAlgolia as jest.MockedFunction<
+  typeof getEventsFromAlgolia
+>;
+const date = new Date('2021-12-28T14:00:00.000Z');
+const renderEvents = async ({
+  searchQuery = '',
+  constraint,
+  pathName,
+  state,
+  isPast,
+}: {
+  searchQuery?: string;
+  constraint: Constraint;
+  pathName: string;
+  state: RecoilState<number>;
+  isPast: boolean;
+}) => {
+  render(
+    <RecoilRoot
+      initializeState={({ reset, set }) => {
+        set(state, Math.random());
+        reset(
+          eventsState({
+            after: '2021-12-28T14:00:00.000Z',
+            searchQuery,
+            filters: new Set<string>(),
+            constraint,
+            currentPage: 0,
+            pageSize: CARD_VIEW_PAGE_SIZE,
+          }),
+        );
+      }}
+    >
+      <Suspense fallback="loading">
+        <Auth0Provider user={{}}>
+          <WhenReady>
+            <MemoryRouter
+              initialEntries={[
+                {
+                  pathname: pathName,
+                },
+              ]}
+            >
+              <Route path={pathName}>
+                <Events
+                  constraint={constraint}
+                  currentTime={date}
+                  past={isPast}
+                />
+              </Route>
+            </MemoryRouter>
+          </WhenReady>
+        </Auth0Provider>
+      </Suspense>
+    </RecoilRoot>,
+  );
+  await waitForElementToBeRemoved(() => screen.queryByText(/loading/i));
+};
+const userPath = (userId: string) =>
+  network({}).users({}).user({ userId }).upcoming({}).$;
+const teamPath = (teamId: string) =>
+  network({}).teams({}).team({ teamId }).upcoming({}).$;
+it.each`
+  description                     | isPast   | constraint                      | getPath     | refreshState        | after                         | before
+  ${'upcoming events by userId '} | ${false} | ${{ type: 'user', id: '1013' }} | ${userPath} | ${refreshUserState} | ${'2021-12-28T13:00:00.000Z'} | ${undefined}
```

Comment:
Why are there variables that don't change? Should there be more tests or should they be removed?

Reply:
remove refreshState and isPast

Reply:
done